sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/utils/CatalogV2Util.scala

```diff
@@ -149,4 +153,11 @@ object CatalogV2Util {
 
     new StructType(newFields)
   }
+
+  def loadTable(catalog: CatalogPlugin, ident: Identifier): Option[Table] =
+    try {
+      Option(catalog.asTableCatalog.loadTable(ident))
+    } catch {
+      case _: NoSuchTableException => None
```

Comment:
So, we return `None` for `NoSuchTableException` only and propagate exceptions for all catalog errors like  `CatalogNotFoundException` from `loadTable` and `AnalysisException` from `asTableCatalog`?

Reply:
Yes.

BTW, I don't think loadTable throws CatalogNotFoundException.